app/src/main/java/com/example/studentarena/MainActivity.java

```diff
@@ -17,38 +25,72 @@
+    private final ProfileFragment fragmentProfile = new ProfileFragment(MainActivity.this);
+    public BottomNavigationView bottomNavigationView;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
+        bottomNavigationView = findViewById(R.id.bottom_navigation);
         bottomNavigationView.setOnItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
             @Override
             public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-                Fragment fragmentToShow = null;
+                Fragment fragmentToShow;
                 switch (item.getItemId()) {
                     case R.id.action_home:
-                        fragmentToShow = new FeedFragment();
+                        fragmentToShow = fragmentHome;
                         break;
                     case R.id.action_compose:
-                        fragmentToShow = new ComposeFragment();
+                        fragmentToShow = fragmentCompose;
                         break;
                     case R.id.action_message:
-                        fragmentToShow = new MessageFragment();
+                        fragmentToShow = fragmentMessage;
                         break;
                     case R.id.action_profile:
-                        fragmentToShow = new ProfileFragment();
+                        fragmentToShow = fragmentProfile;
                         break;
-                    default: break;
+                    default: fragmentToShow = fragmentHome;
                 }
-                if (fragmentToShow != null) {
                     getSupportFragmentManager().beginTransaction().replace(R.id.flContainer, fragmentToShow).commit();
-                }
                 return true;
             }
         });
+        setUserLocation();
         bottomNavigationView.setSelectedItemId(R.id.action_home);
     }
+    private void setUserLocation() {
+        LocationManager mLocationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        long UPDATE_INTERVAL = 10000;  /* 10 secs */
+        long MIN_DISTANCE = 1 * 1609; /* 1 mile */
+        int REQUEST_CODE = 1;
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED &&
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_CODE);
+        }
+
+        mLocationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, UPDATE_INTERVAL, MIN_DISTANCE, new LocationListener() {
```

Comment:
so you're updating location every 10 seconds, but only if the update moves a mile or more? it seems like that cause the location manager to never give you updates, unless i'm misunderstanding

to test this, you could post an item with an address at the office and then go for a nice walk (or go home) and keep the app open to see if the displayed distance changes. or did you already do something like that?